securedrop/tests/test_source.py

```diff
@@ -28,6 +28,11 @@ def setUp(self):
     def tearDown(self):
         utils.env.teardown()
 
+    def test_page_not_found(self):
+        response = self.client.get('/UNKNOWN')
+        self.assertEqual(response.status_code, 404)
+        self.assertIn("locate what you requested", response.data)
+
```

Comment:
To add to what Garrett was saying, you could also add the line
```
        self.assertTemplateUsed('notfound.html')
```
to ensure the correct template is being called.